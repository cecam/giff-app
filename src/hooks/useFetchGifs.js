import {useState, useEffect} from 'react'
import { getGiffs } from '../helpers/getGiffs'


export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGiffs(category)
            .then(gifs => {
                setstate({
                    data: gifs,
                    loading: false
                })
            })
    }, [category])

    return state
}