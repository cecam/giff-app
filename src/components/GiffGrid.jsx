import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

import GiffItem from "./GiffItem"

const GiffGrid = ({category}) => {
    const {data:giffs, loading} = useFetchGifs(category)
    
    return ( 
        <>
            <h3 className="animate__animated animate__fadeInDown">{category}</h3>

            {loading && <p>Loading...</p> }

            <div className="grid">
                {giffs.map(giff => (
                    <GiffItem 
                        key={giff.id} 
                        {...giff} 
                    />
                ))}
            </div>
        </>
    );
}

GiffGrid.propTypes = {
    category: PropTypes.string.isRequired
}
 
export default GiffGrid;