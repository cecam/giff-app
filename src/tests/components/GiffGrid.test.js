import React from 'react';
import { shallow } from 'enzyme'

import GiffGrid from '../../components/GiffGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Tests on <GiffGrid />', () => {
    
    test('should print the component', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GiffGrid category="one punch" />)

        expect( wrapper ).toMatchSnapshot()
    });

    test('should show items with useFetchGifs response', () => {
        const gifs = [{
            id: 'abd',
            url: 'https://localhost:3000',
            title: 'anything'
        },{
            id: '123',
            url: 'https://localhost:3000',
            title: 'anything'
        },]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow( <GiffGrid category="one punch" />)

        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GiffItem').length ).toBe(gifs.length)
    });

})