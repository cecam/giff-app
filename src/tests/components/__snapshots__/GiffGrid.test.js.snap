// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GiffGrid /> should print the component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    one punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="grid"
  />
</Fragment>
`;
